cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -239,6 +240,22 @@ class LeadImpl extends ServerImpl with Lead
     }
   }
 
+  private def checkAuthProvider(props: Properties): Unit = {
+    var authP = props.getProperty(Attribute.AUTH_PROVIDER)
+    if (authP == null) {
+      authP = props.getProperty(Attribute.SERVER_AUTH_PROVIDER)
+    }
```

Comment:
I guess gemxd doesn't support different auth schemes for server and peer authentication. AFAIK, SERVER_AUTH_PROVIDER was added to allow user to only have server authentication and skip peer authentication.